lib/lightning/attempt_service.ex

```diff
@@ -157,6 +255,46 @@ defmodule Lightning.AttemptService do
     |> Repo.one()
   end
 
+  def list_for_rerun_from_start(order_ids) when is_list(order_ids) do
+    attempt_run_numbers_query =
+      from(ar in AttemptRun,
+        join: att in assoc(ar, :attempt),
+        join: r in assoc(ar, :run),
+        where: att.work_order_id in ^order_ids,
+        select: %{
+          id: ar.id,
+          row_num:
+            row_number()
+            |> over(partition_by: att.work_order_id, order_by: r.started_at)
```

Comment:
What is the purpose of windowing these rows, below we're only pulling out the first one - wouldn't it make more sense to pick the first row from a query ordered by started_at?

I also want to suggest that we try order by `r.started_at OR r.inserted_at`, because although rare it's not impossible for a run to have not started (and won't ever be - due to configuration issues).

Reply:
Yeah thanks @midigofrank, it got my attention we were selecting `num_workorders * runs_for_all_attempts`, sorting and numbering them just to drop all but one (for each workorder).

I would have thought dropping the ones that aren't the 'first' before joining would be more efficient. But eh, not a hill I'm willing to die on! :sweat_smile:

After playing around, the way I was expecting it to be is likely much more work than the negligible (if not zero) performance gain. 

Thanks!